### PR TITLE
Editor: Move to next line in run_selection() if nothing selected

### LINKED ISSUE #3115: Automatically advance to the next line after pressing F9
There's already a [question on stackoverflow](http://stackoverflow.com/questions/35611033/how-to-automatically-advance-to-the-next-line-after-f9-in-spyder-ide) that perfectly describes my question:

> Coming from RStudio, I'm familiar with the behavior of the cursor automatically moving to the next line after running the current line. I think this makes sense for data analysis. Does Spyder have something similar?

> I'm familiar with the concept of cell and auto-advancing cell in Spyder. However, I want to maintain a pure Python script (that can be run with python file.py) for portability, and don't want to use something that only makes sense on Spyder.

I know some people in my working group and other institutions that would already like to have such a shortcut for procedural data analysis scripts, plotting, ...

Is there a way to realize that behaviour (without using code cells)?

### PR DESCRIPTION
**Pull request by jitseniesen**
Fixes #3115 

-----
@jitseniesen I rewrote the header to the standard we try to use (`Fixes #<issue number>`)

### CODE REVIEW

**Comment by jitseniesen** [comment]
It seems to me from the log that the AppVeyor failure occured while the test environment is set up so it is not caused by anything in the pull request.

**Comment by goanpeca** [comment]
@jitseniesen Awesome :-), thanks for working on this.

Indeed, the appveyor tests are broken for some weeks now, @ccordoba12 will fix them soon :-)

File: spyderlib/widgets/editor.py [hunk 1]
```diff
@@ -1773,12 +1773,19 @@ def fix_indentation(self, index=None):
 
     #------ Run
     def run_selection(self):
-        """Run selected text or current line in console"""
+        """Run selected text or current line in console
+        If no text is selected, then advance cursor to next line."""
```
Comment (goanpeca) [inline spyderlib/widgets/editor.py]: Minor style comment

Can we turn this into

```python
"""
Run selected text or current line in console. If no text is selected, then
advance cursor to next line.
"""
```

**Comment by goanpeca** [comment]
LGTM :+1:

Any comments @spyder-ide/core-developers

**Comment by blink1073** [comment]
Yep, looks sensible to me.

**Comment by goanpeca** [comment]
**looks sensible to me** WTF?

**Comment by blink1073** [comment]
As in I agree with the approach, but did not actually test it ;).

**Comment by jitseniesen** [comment]
Of course I'm happy to follow the house style (regarding formatting of docstrings), so I changed it. I tried to infer how you wanted it from other docstrings, but obviously failed.

**Comment by goanpeca** [comment]
@jitseniesen thanks for the changes... we do not have an specific style written somewhere, but what I am trying to promote is something in the lines of.

- Docstrings are sentences, they need to end with a `.`.
- Docstrings should have a quick one liner summary and a more lengthy description afterwards.
- Doctsrings respect 80 characters length (PEP enforces a smaller value but mehhh...)
- Doctsrings should ALWAYS use double tripple quotes and ALWAYS be written using a separate line as in:

```python
# As in
def test()
    """
    Hello world.
    """

# Instead of 
def test()
    """Hello world."""
```

This makes the rule easy.... ALWAYS use that instead of one for one liners or another for whatever reason.

What do you think?

**Comment by jitseniesen** [comment]
I think it makes sense for a more-than-one-(wo)man project to have style rules and it does not make much sense to argue long about them, so I'm happy to follow those rules.

**Comment by ccordoba12** [comment]
Thanks for working on this @jitseniesen! Great addition :-)

**Comment by Nodd** [comment]
What is preventing merging this PR ? Everything looks fine ?

**Comment by ccordoba12** [comment]
If you feel it's ok, please merge it :-)

**Pull request merged**